pkg/xds/envoy/clusters/health_check_configurer.go

```diff
@@ -20,19 +24,99 @@ type healthCheckConfigurer struct {
 	healthCheck *mesh_core.HealthCheckResource
 }
 
+func mapUInt32ToInt64Range(value uint32) *envoy_type.Int64Range {
+	return &envoy_type.Int64Range{
+		Start: int64(value),
+		End:   int64(value),
+	}
+}
+
+func mapHttpHeaders(
+	headers []*v1alpha1.HealthCheck_Conf_Http_HeaderValueOption,
+) []*envoy_core.HeaderValueOption {
+	var envoyHeaders []*envoy_core.HeaderValueOption
+	for _, header := range headers {
+		envoyHeaders = append(envoyHeaders, &envoy_core.HeaderValueOption{
+			Header: &envoy_core.HeaderValue{
+				Key:   header.Header.Key.Value,
+				Value: header.Header.Value.Value,
+			},
+			Append: header.Append,
+		})
+	}
+	return envoyHeaders
+}
+
 func (e *healthCheckConfigurer) Configure(cluster *envoy_api.Cluster) error {
 	if e.healthCheck == nil || e.healthCheck.Spec.Conf == nil {
 		return nil
 	}
 	activeChecks := e.healthCheck.Spec.Conf
-	cluster.HealthChecks = append(cluster.HealthChecks, &envoy_core.HealthCheck{
+	healthCheck := envoy_core.HealthCheck{
 		HealthChecker: &envoy_core.HealthCheck_TcpHealthCheck_{
 			TcpHealthCheck: &envoy_core.HealthCheck_TcpHealthCheck{},
 		},
 		Interval:           activeChecks.Interval,
 		Timeout:            activeChecks.Timeout,
 		UnhealthyThreshold: &wrappers.UInt32Value{Value: activeChecks.UnhealthyThreshold},
 		HealthyThreshold:   &wrappers.UInt32Value{Value: activeChecks.HealthyThreshold},
-	})
+	}
+
+	if activeChecks.Tcp != nil {
+		tcpHealthCheck := envoy_core.HealthCheck_TcpHealthCheck{}
+
+		if activeChecks.Tcp != nil {
```

Comment:
I think second `if activeChecks.Tcp != nil {` is not needed

Reply:
That's completely right, my bad 😅

Reply:
Addressed